packages/@react-stately/color/src/useColorAreaState.ts

```diff
@@ -84,36 +82,72 @@ export function useColorAreaState(props: ColorAreaProps): ColorAreaState {
   valueRef.current = color;
 
   let channels = useMemo(() => {
-    if (!xChannel) {
+    let xCh = xChannel;
+    let yCh = yChannel;
+    // determine the color space from the color value
+    let colorSpace = valueRef.current.getColorSpace();
+
+    if (colorSpace === 'rgb') {
+      if (!xChannel) {
+        switch (yChannel) {
+          case 'red':
+          case 'green':
+            xCh = 'blue';
+            break;
+          case 'blue':
+            xCh = 'red';
+            break;
+          default:
+            xCh = 'blue';
+            yCh = 'green';
+        }
+      } else if (!yChannel) {
+        switch (xChannel) {
+          case 'red':
+            yCh = 'green';
+            break;
+          case 'blue':
+            yCh = 'red';
+            break;
+          default:
+            xCh = 'blue';
+            yCh = 'green';
+        }
+      }
+    } else if (!xChannel) {
       switch (yChannel) {
-        case 'red':
-        case 'green':
-          xChannel = 'blue';
+        case 'hue':
+          xCh = colorSpace === 'hsb' ? 'brightness' : 'lightness';
           break;
-        case 'blue':
-          xChannel = 'red';
+        case 'brightness':
+        case 'lightness':
+          xCh = 'saturation';
           break;
         default:
-          xChannel = 'blue';
-          yChannel = 'green';
+          xCh = 'saturation';
+          yCh = colorSpace === 'hsb' ? 'brightness' : 'lightness';
+          break;
       }
     } else if (!yChannel) {
       switch (xChannel) {
-        case 'red':
-          yChannel = 'green';
+        case 'hue':
+          yCh = colorSpace === 'hsb' ? 'brightness' : 'lightness';
           break;
-        case 'blue':
-          yChannel = 'red';
+        case 'brightness':
+        case 'lightness':
+          yCh = 'saturation';
           break;
         default:
-          xChannel = 'blue';
-          yChannel = 'green';
+          xCh = 'saturation';
+          yCh = colorSpace === 'hsb' ? 'brightness' : 'lightness';
+          break;
       }
     }
```

Comment:
maybe the above logic could be abstracted into the Color classes themselves? would make it easier to add new color formats.